src/lib/utils/evm/types.ts

```diff
@@ -1,14 +1,16 @@
 import type { AssetId } from '@shapeshiftoss/caip'
 import type { HDWallet } from '@shapeshiftoss/hdwallet-core'
+import type { MaybeUndefinedFields } from 'lib/types'
 
 type ApproveInput = {
   assetId: AssetId
   spender: string
   amountCryptoBaseUnit: string
   accountNumber: number
+  pubKey: string
 }
 
-export type MaybeApproveInput = Partial<ApproveInput>
+export type MaybeApproveInput = MaybeUndefinedFields<ApproveInput>
```

Comment:
all maybe undefined vs. all maybe optional, which is what we actually wanted here

Reply:
Precisely - that was the initial intent here, and will ensure we pass things explicitly (i.e omitting `pubKey` here would be happy, versus wanting to enforce it as `string | undefined`). There *are* runtime checks but still, better to catch such bugs earlier on!